src/test/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesExampleTest.java

```diff
@@ -236,6 +239,145 @@ public void shouldDemonstrateInteractiveQueries() throws Exception {
     assertTrue(keyValueBean.getKey().startsWith("streams"));
     assertThat(keyValueBean.getValue(), equalTo(3L));
   }
+  
+  @Test
+  public void shouldStartRestApiOnAnyValidHost() throws Exception {
+    // Race condition caveat:  This two-step approach of finding a free port but not immediately
```

Comment:
Hmm.. this is a it worrisome to me, if we cannot have stable examples repo unit test. Note that we have regular cron jobs to run the unit tests and if anything fails we need to look into it. Having an unstable test due to port binding would reduce the sensitivity of the test. Could we improve on fixing it?

Reply:
Hi @guozhangwang, The original tests in the class also use the same way of binding and the comment is mentioned for them as well.
Although this comment serves as a corner case, still would like to know any suggestions for improvement.

Reply:
If we can wrap the search and bind in a retry loop, I think we'd be all good. It's what I've done in the past.

Reply:
This warning comment is pre-existing.  

As for an approach that would help with failures due to the race condition mentioned, the only thing that comes to mind ATM is to do a try/catch around the block of code setting the port and starting the rest proxy server with some reasonable number of retries (2-3?).

Reply:
@bbejeck @vvcephei, The PR contains a try/catch block for binding host and port already.
My bad, missed a test case for that, Will write a test case to elaborate its use.

Reply:
Agree, Adding a try/catch block will add more robustness to the tests.
Will do the change. :+1: